cql3/query_processor.cc

```diff
@@ -119,62 +124,134 @@ query_processor::query_processor(service::storage_proxy& proxy, database& db, qu
 
                     sm::make_derive(
                             "inserts",
-                            _cql_stats.inserts,
-                            sm::description("Counts a total number of CQL INSERT requests.")),
+                            [this] {
+                                return _cql_stats.query_cnt(source_selector::USER, ks_selector::SYSTEM, stm::statement_type::INSERT)
+                                        + _cql_stats.query_cnt(source_selector::USER, ks_selector::NONSYSTEM, stm::statement_type::INSERT)
+                                        + _cql_stats.query_cnt(source_selector::INTERNAL, ks_selector::SYSTEM, stm::statement_type::INSERT)
+                                        + _cql_stats.query_cnt(source_selector::INTERNAL, ks_selector::NONSYSTEM, stm::statement_type::INSERT);
+                            },
+                            sm::description("Counts the total number of CQL INSERT requests.")),
 
                     sm::make_derive(
                             "updates",
-                            _cql_stats.updates,
-                            sm::description("Counts a total number of CQL UPDATE requests.")),
+                            [this] {
+                                return _cql_stats.query_cnt(source_selector::USER, ks_selector::SYSTEM, stm::statement_type::UPDATE)
+                                        + _cql_stats.query_cnt(source_selector::USER, ks_selector::NONSYSTEM, stm::statement_type::UPDATE)
+                                        + _cql_stats.query_cnt(source_selector::INTERNAL, ks_selector::SYSTEM, stm::statement_type::UPDATE)
+                                        + _cql_stats.query_cnt(source_selector::INTERNAL, ks_selector::NONSYSTEM, stm::statement_type::UPDATE);
+                            },
+                            sm::description("Counts the total number of CQL UPDATE requests.")),
 
                     sm::make_derive(
                             "deletes",
-                            _cql_stats.deletes,
-                            sm::description("Counts a total number of CQL DELETE requests.")),
+                            [this] {
+                                return _cql_stats.query_cnt(source_selector::USER, ks_selector::SYSTEM, stm::statement_type::DELETE)
+                                        + _cql_stats.query_cnt(source_selector::USER, ks_selector::NONSYSTEM, stm::statement_type::DELETE)
+                                        + _cql_stats.query_cnt(source_selector::INTERNAL, ks_selector::SYSTEM, stm::statement_type::DELETE)
+                                        + _cql_stats.query_cnt(source_selector::INTERNAL, ks_selector::NONSYSTEM, stm::statement_type::DELETE);
+                            },
+                            sm::description("Counts the total number of CQL DELETE requests.")),
+
+                    sm::make_derive(
+                            "system_reads",
```

Comment:
Please don't use system as a metric name, it should be a keyspace label (see my comment and references on the previous version)

The same goes for the rest of the operations.

make sure that when summing a metric over all labels you don't count the same metric twice